### ISSUE TITLE
Legends and coloring groupings 

### ISSUE THREAD

**Issue by PrincessTheDog** [stemangiola/tidyHeatmap#86]
Hey there! First of all, I love the package! I'm a long time pheatmap user, and I've made some very attractive heatmaps with your package. I have just a few questions:
-When I try to run the example with add_tile(activation, show_legend = FALSE) I get the error `Error in add_tile(., activation, show_legend = FALSE) : 
  unused argument (show_legend = FALSE)`
-I've used the palette_grouping command to color my row groupings. Is there any way to color column groupings? There's an example on the github page for version 1.1.2 under "Multiple groupings and annotations" that has colored groupings, but is no longer reproducible in the latest version.
-Finally, is there anyway to adjust the font size or remove the box around the group labels? 

Thanks!

**Comment by stemangiola** [comment]
Hello @PrincessTheDog ,

> -When I try to run the example with add_tile(activation, show_legend = FALSE) I get the error

try to install from github master and try

```r
tidyHeatmap::pasilla |>
	group_by(location, type) |>
	heatmap(
		.column = sample,
		.row = symbol,
		.value = `count normalised adjusted`,
		show_heatmap_legend = FALSE
	) |>
	add_tile(condition, show_legend = FALSE) |>
	add_tile(activation, show_legend = FALSE)
```




> I've used the palette_grouping command to color my row groupings. Is there any way to color column groupings?

See now update github homepage, try

```r
# Make other groupings
mtcars_tidy_groupings = 
	mtcars_tidy |>
	mutate(property_group = if_else(Property %in% c("cyl", "disp"), "Engine", "Other"))

mtcars_tidy_groupings |> 
	group_by(vs, property_group) |>
	heatmap(
		`Car name`, Property, Value	,
		palette_grouping = list(c("#66C2A5", "#FC8D62"), c("#b58b4c", "#74a6aa"))
	) |>
	add_tile(hp)
```



> -Finally, is there anyway to adjust the font size or remove the box around the group labels?

Unfortunately, no yet but feel free to add a dedicated feature request here on github

**Comment by PrincessTheDog** [comment]
Thanks for the prompt response! Just generated some really lovely heatmaps! I hope my collaborators are ready to be wowed.

**Comment by singjc** [comment]
>> -Finally, is there anyway to adjust the font size or remove the box around the group labels?

>Unfortunately, no yet but feel free to add a dedicated feature request here on github

I was wondering if there's anyway to change the font size by converting the heatmap to a ComplexHeatmap object, and change the gp font size from the object itself? I tried looking, but couldn't find where this gets stored in the ComplexHeatmap object. Would you know where I can find the inner data elements to change this?

- Justin

**Comment by stemangiola** [comment]
Hello, if you mean x,y labels, yes, please have a look to the README

**Comment by singjc** [comment]
Hi sorry, I meant the faceting strip panels, I would like to change the text font size and the box size if possible. I was trying to look into the ComplexHeatmap object (converted from the tidyHeatmap object) itself, to see if I can change it there, but I couldn't find an element that handles the group faceting component.

As an example

![image](https://github.com/stemangiola/tidyHeatmap/assets/32938975/bda9c2e5-f195-4ca5-bde7-51e4c3ece45b)

**Comment by stemangiola** [comment]
please have a look at how complexheatmap handles it,  I might be able to add the feature

**Comment by ggruenhagen3** [comment]
I second this request. All other features of tidyHeatmap are really seamless and easy for me, except for control over the group_by facet, which is something I find myself needing to do often.

**Comment by stemangiola** [comment]
Please have a look at and test the new `annotation_group` function. See if customisation meets your needs.

**Comment by ggruenhagen3** [comment]
I like the new function, but I have a couple of requests for it:
1) An option for the color of the font.
2) An option to turn off the panel entirely. This is because I like to use the splits but don't always need labels and boxes for the groups.
3) An option to remove the border around the boxes.
4) An option to set all these options for each grouping separately similar to how palette_grouping works.

**Comment by stemangiola** [comment]
Glad you like it. 

For customisability, we are more limited by design compared to `ComplexHeatmap`. I tend to be parsimonious with the arguments we add, to keep the interface lean. 

I will close the issue for now, but if there are more requests for customisability from the community, I will happily reopen the issue.

**Comment by ggruenhagen3** [comment]
I understand and this added level of customizability is already much appreciated! Though if you find some time in the future, a single option for the font color (ie similar to how group_label_fontsize currently is) and a single option for border color would be helpful. Thank you again for this easy-to-use package, I love how I can create heatmaps much more easily and quickly than doing it all myself in ComplexHeatmap.

**Comment by ggruenhagen3** [comment]
I've found now that I can control a lot of the things I requested earlier by taking advantage of tidyHeatmap's flexible objects. For example, I can remove borders around the annotation_group boxes by controlling the gpar like so:  `tidy_ht@group_left_annotation$INSERT_NAME_OF_YOUR_GROUP@var_env$gp = gpar(lwd=0)`. I rescind my request for the added features above and thank you again for the great package, I use it all the time!
